strictdoc/backend/sdoc_source_code/comment_parser/marker_lexer.py

```diff
@@ -62,16 +66,23 @@ class GrammarTemplate(Template):
 
 class MarkerLexer:
     @staticmethod
-    def parse(source_input: str, parse_nodes: bool = False) -> ParseTree:
-        if parse_nodes:
+    def parse(
+        source_input: str, parse_nodes: Optional[list[str]] = None
+    ) -> ParseTree:
+        if parse_nodes is not None:
+            grammar_extension = NODE_GRAMMAR_EXTENSION.substitute(
+                ALLOWED_TAGS="|".join(parse_nodes),
+                RESERVED_KEYWORDS=RESERVED_KEYWORDS,
+                RELATION_MARKER_START=RELATION_MARKER_START,
+            )
             start = "(relation_marker | node_field | _NORMAL_STRING_NO_MARKER_NO_NODE | _WS)*"
         else:
+            grammar_extension = ""
             start = "(relation_marker | _NORMAL_STRING_NO_MARKER | _WS)*"
 
         grammar = GRAMMAR.substitute(
-            REGEX_NODE_NAME=REGEX_NODE_NAME,
+            GRAMMAR_EXTENSION=grammar_extension,
```

Comment:
I am not sure if extracting this part of the grammar will not create extra confusion long-term. The argument I have against doing this is that the extracted fragment is still highly coupled to the rest of the grammar, so it is not really a standalone unit being extracted.

Reply:
I did this because, when there are zero allowed tags, what should be the grammar definition of the `node_field` symbol?

I.e., there's no more possible match. I was looking for a lark keyword to put at the right side of a BNF rule that means "no possible match", but afaikt there's no such keyword.  So the symbol should vanish completely from the grammar.  And if it vanishes, all other symbols that are only reachable through  `node_field` should also vanish. So a whole block of python t-string text shall vanish programmatically. I'm quite unfamiliar yet with t-strings, pulling the block out as separate template was the only idea I had.

What would you suggest?